src/script/standard.cpp

```diff
@@ -89,21 +89,45 @@ static constexpr bool IsSmallInteger(opcodetype opcode)
     return opcode >= OP_1 && opcode <= OP_16;
 }
 
+/** Valid number of keys in a multisig are [1,20]. */
+static constexpr bool IsValidMultisigKeyCount(unsigned int n_keys)
+{
+    return n_keys > 0 && n_keys <= MAX_PUBKEYS_PER_MULTISIG;
+}
+
+static bool GetMultisigKeyCount(const CScript& script, CScript::const_iterator& it, unsigned int& count, opcodetype opcode, valtype data)
+{
+    if (IsSmallInteger(opcode)) {
+        count = CScript::DecodeOP_N(opcode);
+    } else {
+        try {
+            count = CScriptNum(data, true).getint();
+        } catch (const scriptnum_error&) {
+            return false;
+        }
+    }
+    if (!IsValidMultisigKeyCount(count)) return false;
```

Comment:
What are your thoughts on just doing: `return IsValidMultisigKeyCount(count);` here?

Reply:
Right, it's cleaner :) Done